compiler/ir/serialization.common/src/org/jetbrains/kotlin/backend/common/serialization/IrModuleDeserializer.kt

```diff
@@ -37,9 +37,9 @@ abstract class IrModuleDeserializer(val moduleDescriptor: ModuleDescriptor) {
     abstract fun deserializeIrSymbol(idSig: IdSignature, symbolKind: BinarySymbolData.SymbolKind): IrSymbol
 
     open fun declareIrSymbol(symbol: IrSymbol) {
-        assert(symbol.isPublicApi) { "Symbol is not public API: ${symbol.descriptor}" }
+        assert(symbol.signature != null && symbol.isPublicApi) { "Symbol is not public API: ${symbol.descriptor}" }
```

Comment:
Here and below, please extract `symbol.signature` to a variable. Also, replace `assert`s  with `require` here, to make use of smart casts to avoid extra `!!`